core/shared/src/main/scala/org/http4s/headers/`X-Forwarded-Host`.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+package headers
+
+object `X-Forwarded-Host` extends HeaderCompanion[`X-Forwarded-Host`]("X-Forwarded-Host") {
+  def apply(host: String, port: Int): `X-Forwarded-Host` = apply(host, Some(port))
+
+  private[http4s] val parser =
+    Host.parser.map(host => apply(host.host, host.port))
+
+  implicit val headerInstance: Header[`X-Forwarded-Host`, Header.Single] =
+    createRendered(xfh => Host(xfh.host, xfh.port))
```

Comment:
Just talking out loud to myself:
I was a bit surprised by the usage of `Host` here but this value is ultimately passed to `Renderer.renderString` and we can reuse the `Host` renderer here since the two headers are ~ the same

Reply:
yeah, that was the idea. I was trying to avoid duplicating logic